hnn_core/network.py

```diff
@@ -328,6 +328,11 @@ class Spikes(object):
         The inner list contains the type of spike (e.g., evprox1
         or L2_pyramidal) that occured at the corresonding time stamp.
         Each gid corresponds to a type via Network::gid_dict.
+    vsoma : dict
+        Dictionary indexed by gids containing somatic voltages
+    t_vec : list
+        List of time points for samples in continuous data.
+        This includes vsoma.
```

Comment:
I know `t_vec` is extensively used in HNN GUI, but the only place I can see it used here is in the example. Other code in this PR that creates the NEURON object, sets up the recording, and copies after the end of simulation might be unnecessary if the example (plotting) recreated t_vec as a numpy array from 0 to `t_stop` with `dt`.
https://github.com/jonescompneurolab/hnn-core/blob/7f142513bd7d42e6dfcd660126551d3962fabb07/examples/plot_simulate_gamma.py#L57

Reply:
I understand your point, however this relies on simulation time always increasing in increments of `dt`. We we sure this is always the case? Especially during parallel runs.

Reply:
Yes, we never use the variable step integration method. I imagine this choice was made early on. More recently, Michael Hines said we might compare the two, so I tried the variable step method and it slows down the simulation. Using ParallelContext does not affect this.

Reply:
Good to know. In that case what do you think of removing the t_vec recording that is passed to the `Dipole` object as well? If it's equivalent to `np.arange()` we may as well reduce the amount of data being passed around.

Reply:
In principle, I agree with using `np.arrange()` instead of, in each case, creating a `t_vec` hoc object, copying with `.to_python()`, and then passing around. This will become a rather common task in `Dipole` and `Network`, so I wonder if this justifies a utility function with `tstop` and `dt` parameters? I know it's just a one-liner, but it would help to keep this process standard (i.e. t[-1]=170.0 vs. only t[-1]=169.75). I don't know where to put it though... @jasmainak?

Reply:
Ideally, the code should be structured so that `times` is "inherited" from other objects without recreating it. For example, if the `Network` object creates it once, then the `Dipole` and `Spikes` object should copy that. That way you make sure there are no bugs ... even if you use variable time etc. It's what we do in MNE, for example when [creating evoked from epochs](https://github.com/mne-tools/mne-python/blob/master/mne/epochs.py#L1038)